deploy/stacks/opensearch.py

```diff
@@ -77,7 +79,7 @@ def __init__(
         self.domain = opensearch.Domain(
             self,
             f'OpenSearchDomain{envname}',
-            domain_name=f'{resource_prefix}-{envname}-domain',
+            domain_name=re.sub(r"[^a-z0-9-]","",f'{resource_prefix}-{envname}-domain')[:28].lower(),
```

Comment:
I believe we should reuse `NamingConventionPattern` &  `build_compliant_name`

Reply:
You are referring to the class in `backend/dataall/utils/naming_convention.py` right? yeep I agree